### DISCUSSION TITLE
Use pyscript to develop an integration ?

### DISCUSSION THREAD

**Discussion by vincegre** [custom-components/pyscript#586] [General]
Hi,

Sorry for the question from a noob in Python. I want to write an integration to control some devices but for now writing an integration is quite above my abilities in Python. Pyscript looks a little better and so is it possible to write something equivalent at an integration with Pyscript ?

Thanks

Vincèn

**Comment by ALERTua** [comment]
While Pyscript has its language limitations, I guess it is still possible to at least write your service and automation factory based on user config. So, basically, this would seem like a typical Home Assistant Integration.
You could describe your idea for more detailed advice.

**Comment by vincegre** [comment]
> While Pyscript has its language limitations, I guess it is still possible to at least write your service and automation factory based on user config. So, basically, this would seem like a typical Home Assistant Integration. You could describe your idea for more detailed advice.

Thanks for your answer. It's an integration to control some pro Audiovisual devices like audio/video matrix or similar devices. Most of them use telnet for their control so it should not be too difficult to implement I guess in Python ? ;)
